### PR TITLE
Python 3 support

### PR DESCRIPTION
**Pull request by takluyver**
Following discussion on PR #32, this adds Python 3 support without any external dependencies. Python 2.6 and above and Python 3.3 and above should work; older versions will not.

The tests are passing on Python 2 and 3; I haven't tried the examples.

### CODE REVIEW

**Comment by sontek** [comment]
To help with this I added a PR to this PR:

https://github.com/takluyver/xlwt/pull/1

That adds tox and pytest support and fixes bugs in a test that wasn't being ran by nose.

**Comment by sontek** [comment]
What is left to get this merged in? I'd love to see it :)

**Comment by sontek** [comment]
With takluyver#2 I've added the badges for pypi, coveralls, and travis.  We should enable coveralls on this repo to get the badge working properly.

**Comment by sontek** [comment]
@cjw296 Beers on me at PyCon if you merge this :)  :beers: :cocktail:

**Comment by collinanderson** [comment]
+1

**Comment by cjw296** [comment]
I'm coming to 3 days of sprints at PyCon with the express intent of merging this.
@sontek, hopefully you and @takluyver can be there to help?

**Comment by sontek** [comment]
@cjw296 we were hoping to get a release prior to pycon. We just ported all our stuff to py3 and want to have it fully in prod running stable releases before pycon.

Could we consider merging this to a develop/feature branch and release a prerelease version people can test out?

Are there specific things in this PR you feel aren't ready that we could help with?

**Comment by takluyver** [comment]
I will be at Pycon, but I'd be happy to get this merged before then. I can think of more interesting things to be working on during the few days of sprints with Pythonistas from around the world.

**Comment by tjanez** [comment]
Hi!

Any progress on this?

**Comment by alexandre** [comment]
up

**Comment by collinanderson** [comment]
(btw if it helps, xlwt-future has py3 support, though beware it doesn't play will with newrelic)

File: xlwt/antlr.py [hunk 1]
```diff
@@ -1657,7 +1650,7 @@ def set(self,bit,on=True):
         if i>=len(self.data):
             d = i - len(self.data) + 1
             for x in xrange(0,d):
-                self.data.append(0L)
+                self.data.append(0)
```
Comment (collinanderson) [inline xlwt/antlr.py]: Maybe it would be safer to try to return long(0) on py2

**Comment by takluyver** [comment]
Woo! :tada: :sparkles: :cake:

**Comment by nicklasb** [comment]
I know what they fix, the blatant lack of back doors. 
Woo!! . :-)

**Comment by cjw296** [comment]
@nicklasb - huh?

**Comment by nicklasb** [comment]
Just kidding. 
Just wrt big commits that do more then they say and that I understand your apprehension. :-)

**Pull request merged**